src/ra_server_proc.erl

```diff
@@ -1555,6 +1551,51 @@ do_state_query(voters, #{cluster := Cluster}) ->
     Vs;
 do_state_query(members, #{cluster := Cluster}) ->
     maps:keys(Cluster);
+do_state_query(members_info, #{cfg := #cfg{id = Self}, cluster := Cluster,
+                 leader_id := Self, query_index := QI, commit_index := CI,
+                 membership := Membership}) ->
+    maps:map(fun(Id, Peer) ->
+                 case {Id, Peer} of
+                     {Self, Peer = #{voter_status := _}} ->
+                         %% For completeness sake, preserve `target`
+                         %% of once promoted leader.
+                         #{next_index => CI+1,
+                          match_index => CI,
+                          query_index => QI,
+                          status => normal,
+                          voter_status => Peer#{membership => Membership}};
```

Comment:
Sould this be the voter status map instead of the full `Peer`?

Reply:
That's what I get for working late night!